src/wasm/kotlin/java/arcs/Particle.kt

```diff
@@ -247,23 +230,21 @@ class Collection<T : Entity<T>>(private val entityCtor: () -> T) : Handle(), Ite
 
     override fun iterator() = entities.values.iterator()
 
-    override fun sync(encoded: String?) {
+    override fun sync(encoded: ByteArray) {
         entities.clear()
-        encoded?.let { add(it) }
+        add(encoded)
     }
 
-    override fun update(added: String?, removed: String?) {
-        added?.let { add(added) }
-        removed?.let {
-            with(StringDecoder(it)) {
-                var num = getInt(":")
-                while (num-- > 0) {
-                    val len = getInt(":")
-                    val chunk = chomp(len)
-                    // TODO: just get the id, no need to decode the full entity
-                    val entity = entityCtor().decodeEntity(chunk)!!
-                    entities.remove(entity.internalId)
-                }
+    override fun update(added: ByteArray, removed: ByteArray) {
+        add(added)
+        with(StringDecoder(removed)) {
+            var num = getInt(':')
+            while (num-- > 0) {
+                val len = getInt(':')
+                val chunk = chomp(len)
+                // TODO: just get the id, no need to decode the full entity
+                val entity = entityCtor().decodeEntity(chunk)!!
```

Comment:
Rather than use `!!`,  it's preferred to use `requireNotNull()`

```kotlin
val entity = requireNotNull(entityCtor().decodeEntity(chunk))
```

Reply:
Done.

("Let's make this common pattern really terse for readability!"  "...no, not there")